testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -4537,6 +4537,27 @@ public void testUpdateWithPredicates()
         }
     }
 
+    @Test
+    public void testUpdateWithPredicatesForVarchars()
+    {
+        if (!hasBehavior(SUPPORTS_UPDATE)) {
+            // Note this change is a no-op, if actually run
+            assertQueryFails("UPDATE nation SET nationkey = nationkey + regionkey WHERE regionkey < 1", MODIFYING_ROWS_MESSAGE);
```

Comment:
if you do add this test, make sure it's run against a "COPY" of the `nation` table. Tests should avoid modifying `tpch` tables since other concurrently running tests rely on them.

Reply:
> Add test `UPDATE nation SET nationkey = nationkey + regionkey WHERE regionkey = 1`

To be honest didn't get it, why to add this condition. The row you pointed is executed in case we run the update statement on connectors, which doesn't support updates.

So we expect to fail with `MODIFYING_ROWS_MESSAGE` on very early stage before even predicate will be resolved.

Reply:
tests added